### PR TITLE
migrate jooby codec v3 to v4

### PR DESCRIPTION
**Pull request by aleks-and**

### CODE REVIEW

File: src/js/components/LogItem/LogItem.jsx [hunk 1]
```diff
@@ -341,133 +351,171 @@ const LogItem = ({
...
-                                    <TypographyMono>
-                                        {`0x${utils.getHexFromNumber(frameParameters.destination, {separator: ''})}`}
-                                    </TypographyMono>
-                                    <TypographyBold>source address</TypographyBold>
-                                    <TypographyMono>
-                                        {`0x${utils.getHexFromNumber(frameParameters.source, {separator: ''})}`}
-                                    </TypographyMono>
-                                    <TypographyBold>message ID</TypographyBold>
-                                    <TypographyMono>{frameParameters.messageId}</TypographyMono>
+                                    {!isNullOrUndefined(frameParameters.type) && (
```
Comment (DeadByElpy) [inline src/js/components/LogItem/LogItem.jsx:354]:
```suggestion
                                    {!!(frameParameters.type) && (
```

?

**Review by DeadByElpy — CHANGES_REQUESTED** [review]

File: src/js/components/LogItem/utils/createLogTitle.jsx [hunk 2]
```diff
@@ -54,26 +32,29 @@ const renderHardwareType = ( hardwareType, commandType ) => {
...
-                    {createMessageDirectionIcon(log.data.commands, log.commandType)}
+                    {createDirectionIcon(log.data.commands[0].command.directionType)}
                     <Box>
                         <Box sx={{minWidth: 0}}>
                             {'commands: '}
                             <HighlightedText>{log.data.commands.length}</HighlightedText>
                             {'; '}
                             {
-                                hasLrc(log.commandType) && log.data.lrc.expected !== log.data.lrc.actual
+                                hasLrc(log.commandType) && ( !actualLrc || !expectedLrc || actualLrc !== expectedLrc )
```
Comment (DarkPark) [inline src/js/components/LogItem/utils/createLogTitle.jsx:48]:
```suggestion
                                hasLrc(log.commandType) && (!actualLrc || !expectedLrc || actualLrc !== expectedLrc)
```

File: src/js/constants.js [hunk 3]
```diff
@@ -1,22 +1,52 @@
...
+    DOWNLINK: 1,
+    UPLINK: 2
+};
+
+export const directionNames = {
+    [directions.DOWNLINK]: 'downlink',
+    [directions.UPLINK]: 'uplink'
+};
+
+export const commands = {
```
Comment (DarkPark) [inline src/js/constants.js:35]: is it really a constant?

File: src/js/components/ParseSection.jsx [hunk 4]
```diff
@@ -121,42 +128,106 @@ const ParseSection = ( {setLogs, hardwareType} ) => {
...
-            } catch ( error ) {
-                parseError = error;
+
+                case COMMAND_TYPE_ANALOG:
+                    try {
+                        direction = Number(parameters.direction);
+                        data = codec.message[directionNames[direction]].fromBytes(
+                            bytes,
+                            {
+                                hardwareType: getHardwareType(hardwareType)
```
Comment (DarkPark) [inline src/js/components/ParseSection.jsx:166]:
```suggestion
                                {hardwareType: getHardwareType(hardwareType)}
```

such simple (and rather short) cases without nesting I propose to write in a single line

**Review by DarkPark — CHANGES_REQUESTED** [review]

File: src/js/components/LogItem/LogItem.jsx (hunk 1 above)
Comment (aleks-and) [reply src/js/components/LogItem/LogItem.jsx]: I removed the function where the value cannot be 0.

File: src/js/components/LogItem/utils/createLogTitle.jsx (hunk 2 above)
Comment (aleks-and) [reply src/js/components/LogItem/utils/createLogTitle.jsx]: fixed

File: src/js/components/ParseSection.jsx (hunk 4 above)
Comment (aleks-and) [reply src/js/components/ParseSection.jsx]: fixed

File: src/js/constants.js (hunk 3 above)
Comment (aleks-and) [reply src/js/constants.js]: Added the file `joobyCodec.js` to the root of the project.

**Review by DeadByElpy — APPROVED** [review]

**Review by DarkPark — APPROVED** [review]

**Pull request merged**